### ISSUE TITLE
Add Option to Add Fields to Registration Form

### ISSUE THREAD

**Issue by Ralkage** [FriendsOfFlarum/masquerade#24]
It has come to my attention that the user bio will be removed entirely from core in beta 8. I can already predict that Masquerade will be one of the top wanted extensions when that time comes to fill in the gap for user bios. So, as a feature request and suggestion, would it be possible to add an option to add the fields created into the sign-up modal? I think it would be perfect and attractive for communities that want user information when the user signs up. Such as age, gender, location, birthday, etc.

**Comment by luceos** [comment]
The sign-up modal isn't that extensible. As long as a full modal replacement is necessary, Masquerade will not offer such functionality. I really don't want to have extension compete for which will replace the modal.

**Comment by Ralkage** [comment]
@luceos could this be revisited now or are there still issues with the Signup modal?

**Comment by luceos** [comment]
Yes good point. Whoever fixes beta 8/9 compatibility and adds this I'll gladly pay for his effort in coffee money.

**Comment by askvortsov1** [comment]
I feel like the "require field completion" for required fields more-or-less covers this, while keeping the signup modal uncluttered, and still well supporting alternate authentication sources (like OAuth or LDAP or SAML).
